cloud-support/custom/paho-openssl/client_sample.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ *
+ * Trust Onboard Paho Client Sample
+ *
+ * Copyright (c) 2019 Twilio, Inc.
+ *
+ * SPDX-License-Identifier:  Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <string>
+#include <iostream>
+
+#include <MQTTClient.h>
+#include "linuxToB/MQTTLinuxToB.h"
+
+volatile int toStop = 0;
+volatile int arrivedcount = 0;
+
+void cfinish(int sig)
+{
+  signal(SIGINT, NULL);
+  toStop = 1;
+}
+
+static void print_usage() {
+  fprintf(stderr, "client_sample <mqtt host> <mqtt port> <signing|available> <serverca.pem> <clientId> <topic>\n");
+  fprintf(stderr, "\n");
+  fprintf(stderr, "Ensure the OPENSSL_CONF environment variable is also set for Trust Onboard OpenSSL engine configuration.\n");
+}
+
+void messageArrived(MQTT::MessageData& md)
+{
+  MQTT::Message &message = md.message;
+
+  printf("Message %d arrived: qos %d, retained %d, dup %d, packetid %d\n",
+      ++arrivedcount, message.qos, message.retained, message.dup, message.id);
+  printf("Payload %.*s\n", (int)message.payloadlen, (char*)message.payload);
+  fflush(stdout);
+}
+
+int main(int argc, const char** argv) {
+  const char* openssl_conf = getenv("OPENSSL_CONF");
```

Comment:
This check probably saves user from some cryptic error messages, but strictly speaking defining `OPENSSL_CONF` is not required, the user can choose other ways to specify the configuration (e.g. editing `/etc/ssl/openssl.cnf`).

I would rather try `ENGINE* eng = ENGINE_by_id("tob_mias")` and check for `eng != nullptr`, then suggest user sets `OPENSSL_CONF`.

Reply:
I like that approach, I'll change this accordingly.